### PR TITLE
Replace input submit with button element for forms

### PR DESCRIPTION
**Pull request by andrew-147**
From the JS standpoint we may come up against the issues in `core/misc/states.es6.js` (specifically for _disabled_ state) and `core/misc/dialog/dialog.ajax.es6.js` (adding button's styles and providing basic functionality). We can revisit this later if there is a need for a patch or some other workaround. But still I prefer using `<button>` while keeping in mind the possibility of issues to occur and tackle them only if there is a reason.

### CODE REVIEW

File: draft_theme.theme [hunk 1]
```diff
@@ -38,3 +38,15 @@ function draft_theme_preprocess_views_view_unformatted(&$variables) {
 //  }
 
 }
+
+/**
+ * Implements hook_preprocess_input().
+ *
```
Comment (T2L) [inline draft_theme.theme]: Remove extra space

File: draft_theme.theme [hunk 2]
```diff
@@ -38,3 +38,15 @@ function draft_theme_preprocess_views_view_unformatted(&$variables) {
...
+ */
+function draft_theme_preprocess_input(&$variables) {
+  // Use <button> element instead of <input type="submit" />
+  // @todo Remove this when added to Drupal core. https://www.drupal.org/node/1671190
+  if ($variables['attributes']['type'] == 'submit') {
```
Comment (T2L) [inline draft_theme.theme]: Use `===`

File: templates/form/input--submit.html.twig [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{#
+/**
+ * @file
+ * Theme override for an 'input' submit form element.
```
Comment (T2L) [inline templates/form/input--submit.html.twig]: Apply the same style as in other ...

File: draft_theme.theme [hunk 4]
```diff
@@ -38,3 +38,15 @@ function draft_theme_preprocess_views_view_unformatted(&$variables) {
...
 
 }
+
+/**
+ * Implements hook_preprocess_input().
+ *
+ */
+function draft_theme_preprocess_input(&$variables) {
+  // Use <button> element instead of <input type="submit" />
+  // @todo Remove this when added to Drupal core. https://www.drupal.org/node/1671190
```
Comment (T2L) [inline draft_theme.theme]: Use `@see` doc tag

File: draft_theme.theme [hunk 5]
```diff
@@ -38,3 +38,15 @@ function draft_theme_preprocess_views_view_unformatted(&$variables) {
 //  }
```
Comment (T2L) [inline draft_theme.theme]: Move to `/preprocess/form.inc`

**Review by T2L — CHANGES_REQUESTED** [review]

File: draft_theme.theme (hunk 4 above)
Comment (andrew-147) [reply draft_theme.theme]: Link to issue should be inside the '@todo' - https://www.drupal.org/docs/develop/coding-standards/api-documentation-and-comment-standards#todo

File: draft_theme.theme (hunk 5 above)
Comment (andrew-147) [reply draft_theme.theme]: Done.

File: draft_theme.theme (hunk 1 above)
Comment (andrew-147) [reply draft_theme.theme]: Removed extra space.

File: draft_theme.theme (hunk 2 above)
Comment (andrew-147) [reply draft_theme.theme]: Fixed.

File: templates/form/input--submit.html.twig (hunk 3 above)
Comment (andrew-147) [reply templates/form/input--submit.html.twig]: Amended docblock.

**Review by T2L — APPROVED** [review]

**Pull request merged**